src/hotspot/share/opto/compile.cpp

```diff
@@ -2458,6 +2458,17 @@ void Compile::Optimize() {
     if (failing()) return;
   }
 
+  // Lower nodes and match backend-specific patterns with Ideal nodes
+  {
+    TracePhase tp("lower", &timers[_t_lower]);
+    print_method(PHASE_BEFORE_LOWERING, 3);
```

Comment:
Isn't `BEFORE_LOWERING` the same as `AFTER_BARRIER_EXPANSION` right above?

Reply:
This is a fair point, I added it originally since macro expansion also printed before and after but I see now that there is some extra logic that isn't covered before macro expansion. I think it makes sense to remove it.